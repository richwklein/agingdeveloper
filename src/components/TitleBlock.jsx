import React from "react";
import {Box, Typography} from "@mui/material";
import PropTypes from "prop-types";

const TitleBlock = ({title, subtitle}) => {
  return (
    <Box component="header" sx={{mt: 1}}>
      <Typography variant="h3">{title}</Typography>
      {subtitle &&
        <Typography variant="subtitle" component={Box}>
          {subtitle}
        </Typography>
      }
    </Box>
  );
};
TitleBlock.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
};

export default TitleBlock;
