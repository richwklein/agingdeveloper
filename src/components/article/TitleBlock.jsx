import React from "react";
import {Box, Typography} from "@mui/material";
import PropTypes from "prop-types";

const TitleBlock = ({title, description}) => {
  return (
    <Box
      sx={{
        marginTop: 1,
      }}>
      <Typography variant="h2">{title}</Typography>
      <Typography variant="subtitle" component={Box}>
        {description}</Typography>
    </Box>
  );
};
TitleBlock.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default TitleBlock;
