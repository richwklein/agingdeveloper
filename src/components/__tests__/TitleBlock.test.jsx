import React from "react";
import {cleanup, render} from "@testing-library/react";
import TitleBlock from "../TitleBlock";

describe("TitleBlock", () => {
  afterEach(cleanup);

  const renderComponent = () => render(
      <TitleBlock
        title= "Tesing Gatsby with Jest"
        subtitle="You can test with Jest with the best of them." />,
  );

  test("that it renders correctly.", () => {
    const {container} = renderComponent();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("that it renders without a subtitle.", () => {
    const {container} = render(<TitleBlock title="without subtitle" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
