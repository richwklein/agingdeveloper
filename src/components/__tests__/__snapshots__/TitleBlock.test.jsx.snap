// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBlock that it contains a subtitle. 1`] = `
<header
  class="MuiBox-root css-164r41r"
>
  <h3
    class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
  >
    Testing Gatsby with Jest
  </h3>
</header>
`;
